extensions/ql-vscode/test/vscode-tests/minimal-workspace/queries-panel/query-discovery.test.ts

```diff
@@ -16,31 +19,63 @@ describe("QueryDiscovery", () => {
     expect(workspace.workspaceFolders?.length).toEqual(1);
   });
 
+  function mockResolveQueryByLanguage(
+    dataFn: (queryDir: string) => {
+      byLanguage?: Record<string, string[]>;
+      noDeclaredLanguage?: string[];
+      multipleDeclaredLanguages?: string[];
+    },
+  ) {
+    return jest.fn().mockImplementation((queryDir: Uri) => {
+      const data = dataFn(queryDir.fsPath);
+      const value: QueryInfoByLanguage = {
+        byLanguage: Object.keys(data.byLanguage || {}).reduce((result, key) => {
+          result[key] = queriesArrayToRecord(data.byLanguage![key]);
+          return result;
+        }, {} as QueryInfoByLanguage["byLanguage"]),
+        noDeclaredLanguage: queriesArrayToRecord(data.noDeclaredLanguage || []),
```

Comment:
We could move the `|| []` into the declaration of `queriesArrayToRecord` by adding `queries: string[] = []`. I think this would make this a bit shorter.